### ISSUE TITLE
Bug: Vue SFC with lang tsx does not parse as tsx and with wrong with "Parsing error: '>' expected"

### ISSUE THREAD

**Issue by zjc2782171149** [typescript-eslint/typescript-eslint#10757]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

.vue file can not show on playground and the version is Vue3

### Repro Code

```TypeScript
<template>
  <TestJSX />
</template>

<script lang="tsx" setup>
const TestJSX = () => {
  return (
    <div class="123">
      123
      <div class="234">
        234
        <div class="345">
          345
        </div>
      </div>
    </div>
  );
}
</script>
```

### ESLint Config

```javascript
# .eslintrc.yaml
env:
  browser: true
  es2021: true
extends:
  - eslint:recommended
  - plugin:vue/vue3-essential
  - plugin:@typescript-eslint/recommended
overrides: []
parser: vue-eslint-parser
parserOptions:
  parser: '@typescript-eslint/parser'
  project:
    - './tsconfig.eslint.json'
  ecmaVersion: latest
  ecmaFeatures:
    jsx: true
  sourceType: module
  extraFileExtensions:
    - '.vue'
plugins:
  - vue
  - '@typescript-eslint'
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - error
    - single
  semi:
    - error
    - always
  max-len:
    - off
  'vue/html-indent':
    - error
    - 2
  'vue/multi-word-component-names':
    - off
  'vue/attributes-order':
    - warn
  '@typescript-eslint/ban-ts-comment': off
  '@typescript-eslint/no-misused-promises': off
```

### tsconfig

```jsonc
{
  // tsconfig.eslint.json
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "strict": true,
    "jsx": "preserve",
    "jsxImportSource": "vue",
    "sourceMap": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "esModuleInterop": true,
    "lib": ["ESNext", "DOM"],
    "skipLibCheck": true,
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": [
    "src/**/*.ts",
    "src/**/*.d.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "server/**/*.ts",
    "tests/**/*.ts",
    "jest.config.ts",
    "vite.config.ts",
    "auto-imports.d.ts",
    "components.d.ts"
  ]
}
```

### Expected Result

No issues parsing .vue files that have lang="tsx"
For example, if I run "pnpm exec eslint 'src/views/user/tags/content.vue' --cache --fix" in my VSCode terminal, there have not any errors and show the blue color.

<img width="748" alt="Image" src="https://github.com/user-attachments/assets/7a7cd44f-7c07-4673-99aa-2392253abaa6" />

### Actual Result

  8:9  error  Parsing error: '>' expected

### Additional Info

  eslint:cli-engine Lint /Users/src/views/user/tags/content.vue +0ms
  eslint:linter Linting code for /Users/project/src/views/user/tags/content.vue (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/src/views/user/tags/content.vue +0ms
  eslint:linter Apply the processor: 'vue/.vue' +0ms
  eslint:linter A code block was found: '(unnamed)' +0ms
  eslint:linter Parsing: /Users/src/views/user/tags/content.vue +2ms
  eslint:linter Parsing error: '>' expected.
  eslint:linter SyntaxError: '>' expected.
    at ParseError.normalize (/Useresktop/node_modules/.pnpm/vue-eslint-parser@9.4.3_eslint@8.57.0/node_modules/vue-eslint-parser/index.js:98:20)
    at parseScriptFragmentWithOption (/Usersp/proje/node_modules/.pnpm/vue-eslint-parser@9.4.3_eslint@8.57.0/node_modules/vue-eslint-parser/index.js:895:33)
    at parseScriptFragment (/Users/Desktop/projects/node_modules/.pnpm/vue-eslint-parser@9.4.3_eslint@8.57.0/node_modules/vue-eslint-parser/index.js:884:12)
    at parseScriptElement (/Users/Desktop/projects/node_modules/.pnpm/vue-eslint-parser@9.4.3_eslint@8.57.0/node_modules/vue-eslint-parser/index.js:1175:20)
    at parseAsSFC (/Users/Desktop/projects/node_modules/.pnpm/vue-eslint-parser@9.4.3_eslint@8.57.0/node_modules/vue-eslint-parser/index.js:6442:18)
    at Object.parseForESLint (/Users/Desktop/projects/node_modules/.pnpm/vue-eslint-parser@9.4.3_eslint@8.57.0/node_modules/vue-eslint-parser/index.js:6402:53)
    at parse (/Users/Desktop/projects/node_modules/.pnpm/eslint@8.57.0/node_modules/eslint/lib/linter/linter.js:834:22)
    at Linter._verifyWithoutProcessors (/Users/Desktop/projects/node_modules/.pnpm/eslint@8.57.0/node_modules/eslint/lib/linter/linter.js:1311:33)
    at /Users/Desktop/projects/node_modules/.pnpm/eslint@8.57.0/node_modules/eslint/lib/linter/linter.js:1913:29
    at Array.map (<anonymous>) +8s
  eslint:linter Generating fixed text for /Users/src/views/user/tags/content.vue (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:file-enumerator Complete iterating files: ["src/views/user/tags/content.vue"] +9s
  eslint:cli-engine Linting complete in: 9024ms +8s

/Users/src/views/user/tags/content.vue
  8:9  error  Parsing error: '>' expected

✖ 1 problem (1 error, 0 warnings)

**Comment by bradzacher** [comment]
We do not support Vue files ourselves. You'll need tot alk to the Vue parser owners for support. (vue-eslint-parser)

**Comment by zjc2782171149** [comment]
Ok, thank you for your reply and I am going to ask Vue parser owners for support.
